application/app/Http/Controllers/InstitutionUserImportController.php

```diff
@@ -150,4 +146,12 @@ private function getCsvContentValidator(UploadedFile $file): CsvContentValidator
                 'personal_identification_code', 'name', 'email', 'phone', 'department', 'role',
             ])->setRules((new ImportUsersCsvRowValidationRequest)->rules());
     }
+
+    private function isExistingInstitutionUser(string $pin): bool
+    {
+        return InstitutionUser::query()->whereRelation(
+            'user',
+            'personal_identification_code', $pin
+        )->withGlobalScope('policy', InstitutionUserPolicy::scope())->exists();
+    }
```

Comment:
**[blocker]** Hmm, not sure whether this is an issue or not –you know better since you worked on this user import, so you can decide. 

Institution users can also be archived and deactivated. Users who are either archived or deactivated will be excluded from the default scope. See: https://github.com/keeleinstituut/tv-authorization/blob/13ff6059cfae536379bd387f7f7cf970f0c35787/application/app/Models/InstitutionUser.php#L76-L81

So right now, this method will ignore such existing institution users. If I understand correctly, the "next" endpoint called after validation will be user import – that will probably fail, because it will try to create an institution user that already exists (but is either archived or deactivated).

Reply:
@DanBondarenko thank you for pointing it out, I fixed the issue and pushed the changes, please take a look

Reply:
Thank you! LGTM :)